tests/lit-tests/ptr-assign-inst.ispc

```diff
@@ -0,0 +1,26 @@
+// This test checks that we can properply instantiate a template function at
+// the moment when we assign its address to a function pointer.
+
+// RUN: %{ispc} --target=host --nostdlib %s --emit-llvm-text -o - 2>&1 | FileCheck %s
+
+float same_exp(float x);
+uniform int same_exp(uniform int x);
+
+typedef float (*FuncType)(float);
+typedef uniform int (*FuncType2)(uniform int);
+
+template <typename T> T same(T x) { return same_exp(x); }
+
+// CHECK-LABEL: @foo___vyf
+// CHECK-DAG: call <{{.*}}> @same_exp___vyf({{.*}})
+float foo(float a) {
+    uniform FuncType func_ptr = same;
```

Comment:
Looks like the intention here is to check `varying float` case so `uniform` keyword is not needed?

Reply:
This case tests instantiation with a varying float argument, creating a uniform pointer to this instantiation. Although, I have also added a test with a varying function pointer.